pkgs/development/tools/xcbuild/default.nix

```diff
@@ -39,5 +39,6 @@ in stdenv.mkDerivation rec {
     rmdir $out/usr
   '';
 
-  buildInputs = [ cmake zlib libxml2 libpng CoreServices CoreGraphics ImageIO ];
-}
\ No newline at end of file
+  buildInputs = [ cmake zlib libxml2 libpng ninja ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ CoreServices CoreGraphics ImageIO ];
```

Comment:
I can explain that! Nothing in that command line says what target to build for, so it's defaulting to the host. And `-arch` is a Darwin-specific flag, and `__unused` is a Darwin extension, hence the errors.

The way xcbuild / Xcode says what platform to build for is through the `-m<platform>-version-min=` flag, which both sets the deployment target and the target triple (usually set by `-target <triple>`). Those are controlled by some options from the platform's `Info.plist`.

Here's what it should be to build for macOS:

    DefaultProperties = {
        DEPLOYMENT_TARGET_SETTING_NAME = "MACOSX_DEPLOYMENT_TARGET";
        DEPLOYMENT_TARGET_CLANG_ENV_NAME = "MACOSX_DEPLOYMENT_TARGET";
        DEPLOYMENT_TARGET_CLANG_FLAG_NAME = "macosx-version-min";
    };